zerver/lib/actions.py

```diff
@@ -967,20 +972,27 @@ def get_recipient_info(recipient: Recipient,
             for row in subscription_rows
         ]
 
+        def should_send(setting: str, row: Dict[str, Any]) -> bool:
+            if not row['in_home_view']:
+                return False
+            if row[setting] is not None:
```

Comment:
Let's add a comment explaining this:
"""
This implements the structure that the UserProfile stream notification settings are defaults, which can be overridden by the Stream-level settings (if those values are not null).  
"""